includes/FileDeleteForm.php

```diff
@@ -134,6 +134,9 @@ public static function doDelete( &$title, &$file, &$oldimage, $reason, $suppress
 			global $wgUser;
 			$user = $wgUser;
 		}
+		global $wgMemc;
+		$redirKey = wfMemcKey( 'redir', $title->getPrefixedText() );
+		$wgMemc->delete( $redirKey );
```

Comment:
Using the `FileDeleteComplete` hook to do this would be preferred over modifying the core MW file and having this code know about an extension-specific Memcache key.